scipy/datasets/tests/test_data.py

```diff
@@ -71,3 +63,32 @@ def test_electrocardiogram(self):
         # hash check
         assert _has_hash(os.path.join(data_dir, "ecg.dat"),
                          registry["ecg.dat"])
+
+
+def test_clear_cache():
+    # Use Dummy path
+    dummy_basepath = "dummy_cache_dir"
+    os.makedirs(dummy_basepath, exist_ok=True)
+
+    # Create three dummy dataset files
+    dummy_registry = {}
+    for i in range(3):
+        dataset_name = f"data{i}.dat"
+        dummy_registry[dataset_name] = hash(dataset_name)
+        with open(os.path.join(dummy_basepath, dataset_name), 'a'):
+            pass
+
+    # remove single dataset file from cache dir
+    _clear_cache(datasets=["data0.dat"], cache_dir=dummy_basepath,
+                 data_registry=dummy_registry)
+    assert len(os.listdir(dummy_basepath)) == len(dummy_registry) - 1
+
+    # wrong dataset name should raise ValueError
+    with pytest.raises(ValueError):
+        _clear_cache(datasets=["data5.dat"], cache_dir=dummy_basepath,
+                     data_registry=dummy_registry)
+
+    # remove all dataset cache
+    _clear_cache(datasets=None, cache_dir=dummy_basepath,
+                 data_registry=dummy_registry)
+    assert not os.path.exists(dummy_basepath)
```

Comment:
When I run this test locally I see a bunch of warning spam like this:

```
Exception ignored in: <_io.FileIO name=0 mode='rb' closefd=True>
ResourceWarning: unclosed file <_io.TextIOWrapper name=0 mode='r' encoding='UTF-8'>
```

Reply:
This looks like an open file, which should not happen since we used a context manager to create those test data files. I cannot reproduce this on MacOS or Ubuntu locally, can you share a bit more on how to reproduce this and maybe some env details?

In any case, the `with` statement for writing files is now gone because we use pytest `tmp_path` `path` object to write the tmp dataset files. Maybe it is worth trying again with the updated test.

Reply:
This seems to be resolved, test output is clean for me.